src/core/etl/src/Flow/ETL/Rows.php

```diff
@@ -226,13 +212,7 @@ public function first() : Row
      */
     public function flatMap(callable $callable) : self
     {
-        $rows = [];
-
-        foreach ($this->rows as $row) {
-            $rows[] = $callable($row);
-        }
-
-        return new self(...\array_merge(...$rows));
+        return new self(...\array_merge(...\array_map($callable, $this->rows)));
```

Comment:
afaik on bigger datasets, foreach is faster than array_map, I believe that if you look into commits history you will notice that it was initially done like that, but then I reverted it into foreach - array_map looks to be more memory efficient but slower than foreach

Reply:
Reverted as with more revs count in benchmark it was slower, while giving ~2% memory boost.